pkg/services/services.go

```diff
@@ -24,143 +24,281 @@ import (
-			servicequeue.NewService1(bitails.ServiceName, bitailsService.GetChainHeaderByHeight),
+			"ChainHeaderByHeight",
+			namedFuncsToServices1(
+				applyModifierIfExists(options.ChainHeaderByHeightModifier,
+					collectSingleMethodImplementations(allImplementations, func(it Implementation) ChainHeaderByHeightFunc {
+						return it.ChainHeaderByHeight
+					})))...,
 		),
 
 		getStatusForTxIDsServices: servicequeue.NewQueue1(
 			logger,
 			"GetStatusForTxIDs",
-			servicequeue.NewService1(whatsonchain.ServiceName, wocService.GetStatusForTxIDs),
-			servicequeue.NewService1(bitails.ServiceName, bitailsService.GetStatusForTxIDs),
+			namedFuncsToServices1(
+				applyModifierIfExists(options.GetStatusForTxIDsModifier,
+					collectSingleMethodImplementations(allImplementations, func(it Implementation) GetStatusForTxIDsFunc {
+						return it.GetStatusForTxIDs
+					})))...,
 		),
 
 		getUtxoStatusServices: servicequeue.NewQueue2(
 			logger,
 			"GetUtxoStatus",
-			servicequeue.NewService2(whatsonchain.ServiceName, wocService.GetUtxoStatus),
+			namedFuncsToServices2(
+				applyModifierIfExists(options.GetUtxoStatusModifier,
+					collectSingleMethodImplementations(allImplementations, func(it Implementation) GetUtxoStatusFunc {
+						return it.GetUtxoStatus
+					})))...,
 		),
 
 		isUtxoServices: servicequeue.NewQueue2(
 			logger,
 			"IsUtxo",
-			servicequeue.NewService2(whatsonchain.ServiceName, wocService.IsUtxo),
+			namedFuncsToServices2(
+				applyModifierIfExists(options.IsUtxoModifier,
+					collectSingleMethodImplementations(allImplementations, func(it Implementation) IsUtxo {
+						return it.IsUtxo
+					})))...,
+		),
+
+		bsvExchangeRateServices: servicequeue.NewQueue(
+			logger,
+			"BsvExchangeRate",
+			namedFuncsToServices(
+				applyModifierIfExists(options.BsvExchangeRateModifier,
+					collectSingleMethodImplementations(allImplementations, func(it Implementation) BsvExchangeRateFunc {
+						return it.BsvExchangeRate
+					})))...,
 		),
 	}
+
+	walletServices.logActiveServices()
+	return walletServices
+}
+
+func (s *WalletServices) logActiveServices() {
+	if !s.logger.Enabled(context.Background(), slog.LevelDebug) {
+		return
+	}
+
+	type loggable interface {
+		GetNames() (methodName string, serviceNames []string)
+	}
+
+	services := []loggable{
+		&s.rawTxServices,
+		&s.postBEEFServices,
+		&s.merklePathServices,
+		&s.findChainTipHeaderServices,
+		&s.isValidRootForHeightServices,
+		&s.currentHeightServices,
+		&s.getScriptHashHistoryServices,
+		&s.hashToHeaderServices,
+		&s.chainHeaderByHeightServices,
+		&s.getStatusForTxIDsServices,
+		&s.getUtxoStatusServices,
+		&s.isUtxoServices,
+		&s.bsvExchangeRateServices,
+	}
+
+	logAttrs := slices.Map(services, func(e loggable) any {
+		methodName, serviceNames := e.GetNames()
+		return slog.String(methodName, strings.Join(serviceNames, ","))
+	})
+
+	s.logger.Debug("active service methods", logAttrs...)
```

Comment:
We have convention that the log message starts with capital letter, and a little change in message
```suggestion
	s.logger.Debug("Active services by methods", logAttrs...)
```

Reply:
✅ Done